File: frontend/src/utils/validators.ts
export function validateResumeText(text: string): string | null {
  if (!text.trim()) return 'Resume text is required';
  if (text.length < 100) return 'Resume must be at least 100 characters';
  if (text.length > 15000) return 'Resume must be under 15,000 characters';
  return null;
}

export function validateJDText(text: string): string | null {
  if (!text.trim()) return 'Job description is required';
  if (text.length < 50) return 'Job description must be at least 50 characters';
  if (text.length > 8000) return 'Job description must be under 8,000 characters';
  return null;
}

File: backend/app/utils/validators.py
from app.utils.exceptions import (
    ResumeEmptyError,
    JDEmptyError,
    InputTooShortError,
    InputTooLongError,
)
from app.config import get_settings


def validate_resume_text(text: str) -> str:
    text = text.strip()
    if not text:
        raise ResumeEmptyError()
    settings = get_settings()
    if len(text) < settings.min_resume_length:
        raise InputTooShortError(
            f"Resume must be at least {settings.min_resume_length} characters (got {len(text)})",
            details={"min_length": settings.min_resume_length, "actual_length": len(text)},
        )
    if len(text) > settings.max_resume_length:
        raise InputTooLongError(
            f"Resume must be under {settings.max_resume_length} characters (got {len(text)})",
            details={"max_length": settings.max_resume_length, "actual_length": len(text)},
        )
    return text


def validate_jd_text(text: str) -> str:
    text = text.strip()
    if not text:
        raise JDEmptyError()
    settings = get_settings()
    if len(text) < settings.min_jd_length:
        raise InputTooShortError(
            f"Job description must be at least {settings.min_jd_length} characters (got {len(text)})",
            details={"min_length": settings.min_jd_length, "actual_length": len(text)},
        )
    if len(text) > settings.max_jd_length:
        raise InputTooLongError(
            f"Job description must be under {settings.max_jd_length} characters (got {len(text)})",
            details={"max_length": settings.max_jd_length, "actual_length": len(text)},
        )
    return text
